.github/workflows/ci.yml

```diff
@@ -38,10 +43,53 @@ jobs:
           python -m pip install -r documentation/library_guide/requirements.txt
       - name: Build documentation
         run: |
-          mkdir html 
+          mkdir html
           sphinx-build -b html documentation/library_guide/ html/
       - name: Archive build directory
         uses: actions/upload-artifact@v1
         with:
             name: onedpl-html-docs-${{ env.GITHUB_SHA_SHORT }}
             path: html
+
+  linux-testing:
+    name: ${{ matrix.os }}_${{ matrix.cxx_compiler }}_cxx${{ matrix.std }}_${{ matrix.build_type }}
+    runs-on: ['${{ matrix.os }}']
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-18.04
+            cxx_compiler: g++
+            std: 17
+            build_type: release
+    steps:
+      - uses: actions/checkout@v2
+      - name: Configure apt repositories
+        run: |
+          wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+          sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+          rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+          echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+          sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
+      - name: Install oneTBB
+        run: sudo apt install -y intel-oneapi-tbb-devel
+      - name: Install oneAPI Compiler
+        if: matrix.cxx_compiler == 'dpcpp' || matrix.cxx_compiler == 'icpx'
+        run: sudo apt install -y intel-oneapi-compiler-dpcpp-cpp
+      - name: Run testing
+        shell: bash
+        run: |
+          set -x
+          sudo rm -rf /opt/intel/oneapi/dpl || true
+          source /opt/intel/oneapi/setvars.sh intel64
+          if [[ "${{ matrix.build_type }}" == "debug" ]]; then
+            TEST_TIMEOUT=${TEST_TIMEOUT_DEBUG}
+          else
+            TEST_TIMEOUT=${TEST_TIMEOUT_RELEASE}
+          fi
+          mkdir build && cd build
+          cmake -DCMAKE_CXX_STANDARD=${{ matrix.std }} -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+            -DCMAKE_CXX_COMPILER=${{ matrix.cxx_compiler }} -DONEDPL_BACKEND=tbb -DONEDPL_DEVICE_TYPE=HOST ..
```

Comment:
Do you think it could be possible to run testing with `dpcpp` backend on CPU in future? If yes, it makes sense to move BACKEND and DEVICE_TYPE to matrix.

Reply:
I checked, yes it is possible. `backend` and `device_type` added to matrix.